### PR TITLE
#171 - Support for outgoing headers and request query in ClientSlice

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #171 
Added support for outgoing headers and request query in `ClientSlice`

### CODE REVIEW

File: src/test/java/com/artipie/docker/proxy/ClientSliceIT.java [hunk 1]
```diff
@@ -116,8 +125,9 @@ void getManifestByTag() {
 
     @Test
     void getManifestNotFound() {
+        final ClientSlice slice = new ClientSlice(this.client, "mcr.microsoft.com");
         final RepoName name = new RepoName.Valid("dotnet/core/runtime");
-        final ProxyManifests manifests = new ProxyManifests(this.slice, name);
+        final ProxyManifests manifests = new ProxyManifests(slice, name);
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/docker/proxy/ClientSliceIT.java:130]: @olegmoz also consider extracting code to create `ProxyManifests` instance into method: now `ClientSlice`, `RepoName` and `ProxyManifests` instances are created in 3 test methods in the same way.

File: src/main/java/com/artipie/docker/proxy/ClientSlice.java [hunk 2]
```diff
@@ -88,8 +88,12 @@ public Response response(
                 .setHost(this.host)
                 .setPort(ClientSlice.HTTPS_PORT)
                 .setPath(req.uri().getPath())
+                .setCustomQuery(req.uri().getRawQuery())
                 .toString()
         ).method(req.method().value()).followRedirects(true);
+        for (final Map.Entry<String, String> header : headers) {
```
[the remark this answers is not in the document]
Comment (olegmoz) [reply src/main/java/com/artipie/docker/proxy/ClientSlice.java:94]: @olenagerasimova do not see how `forEach` is better here. `headers` is an iterable and not a stream, replacing `for` with `forEach` will be more verbose

File: src/test/java/com/artipie/docker/proxy/ClientSliceIT.java (hunk 1 above)
Comment (olegmoz) [reply src/test/java/com/artipie/docker/proxy/ClientSliceIT.java:130]: @olenagerasimova extracted the methods, thanks

File: src/main/java/com/artipie/docker/proxy/ClientSlice.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (olenagerasimova) [reply src/main/java/com/artipie/docker/proxy/ClientSlice.java:94]: @olegmoz hm, iterable has `forEach` method and 
```
headers.forEach(entry -> request.header(entry.getKey(), entry.getValue()));
```
does not seem more verbose to me, but `for` loop is also fine.

**Pull request merged**